File: python/예상대진표.py
def solution(n,a,b):
    answer = 1

    abigger = False
    if a > b:
        abigger = True

    i = n
    while i != 1:
        if abigger and a%2 == 0 and a-1 == b:
            return answer
        elif not abigger and b%2 == 0 and b-1 == a:
            return answer
        if a % 2 == 0:
            a /= 2
        else:
            a = a//2+1
        if b % 2 == 0:
            b /= 2
        else:
            b = b//2+1
        answer += 1
        i /= 2

print(solution(8, 4, 7))

File: 예상대진표.js
function solution(n, a, b) {
    let answer = 1;
    while (true) {
        if (Math.ceil(a / 2) === Math.ceil(b / 2)) return answer;
        if (a % 2 === 0) a /= 2;
        else a = Math.floor(a / 2 + 1);
        if (b % 2 === 0) b /= 2;
        else b = Math.floor(b / 2 + 1);
        answer++;
    }
}

console.log(solution(64, 4, 5));